proof_ipa.go

```diff
@@ -102,16 +95,22 @@ func (sd StateDiff) Copy() StateDiff {
 	ret := make(StateDiff, len(sd))
 	for i := range sd {
 		copy(ret[i].Stem[:], sd[i].Stem[:])
-		ret[i].SuffixDiffs = make([]SuffixStateDiff, len(sd[i].SuffixDiffs))
-		for j := range sd[i].SuffixDiffs {
-			ret[i].SuffixDiffs[j].Suffix = sd[i].SuffixDiffs[j].Suffix
-			if sd[i].SuffixDiffs[j].CurrentValue != nil {
-				ret[i].SuffixDiffs[j].CurrentValue = &[32]byte{}
-				copy((*ret[i].SuffixDiffs[j].CurrentValue)[:], (*sd[i].SuffixDiffs[j].CurrentValue)[:])
+		ret[i].Suffixes = make([]byte, len(sd[i].Suffixes))
+		copy(ret[i].Suffixes, sd[i].Suffixes)
+		ret[i].Current = make([][]byte, len(sd[i].Current))
+		for j := range sd[i].Current {
+			if len(sd[i].Current[j]) == 0 {
+
+			} else {
+				copy(ret[i].Current[j], sd[i].Current[j])
```

Comment:
I think there's a bug here? We never allocated `ret[i].Current[j]` so it's an empty slice. The `copy` will copy the number of bytes being minimum between  the two parameters, thus 0.

Reply:
Note that in `SerializeProof` an `append` is done, which is correct since it will grow as needed compared to this `copy`.